esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6335,27 +6388,40 @@ bool CWsDfuEx::onDFUFilePublish(IEspContext &context, IEspDFUFilePublishRequest
         if (isEmptyString(recordDefinition))
              throw MakeStringException(ECLWATCH_INVALID_INPUT, "No ECLRecordDefinition defined.");
 
-        StringBuffer userId, oldFileName, newFileName, clusterName, groupName;
-        const char *pGroupName = strchr(fileId, DFUFileIdSeparator);
-        if (!pGroupName)
-            throw MakeStringException(ECLWATCH_INVALID_INPUT, "Failed to read Group Name from FileId '%s'.", fileId);
+        StringArray fileIdItems;
+        fileIdItems.appendList(fileId, DFUFileIdSeparator);
+        if (fileIdItems.length() < 3)
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid FileId '%s'.", fileId);
 
-        groupName.append(pGroupName - fileId, fileId);
-        const char *pClusterName = strchr(++pGroupName, DFUFileIdSeparator);
-        if (!pClusterName)
-            throw MakeStringException(ECLWATCH_INVALID_INPUT, "Failed to read Cluster Name from FileId '%s'.", fileId);
-        clusterName.append(pClusterName - pGroupName, pGroupName);
+        const char *groupName = fileIdItems.item(0);
+        const char *clusterName = fileIdItems.item(1);
+        const char *oldFileName = fileIdItems.item(2);
+        if (isEmptyString(groupName))
+             throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid FileId: empty groupName.");
+        if (isEmptyString(clusterName))
+             throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid FileId: empty clusterName.");
+        if (isEmptyString(oldFileName))
+             throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid FileId: empty FileName.");
 
-        oldFileName.set(++pClusterName);
-        newFileName.set(pClusterName);
+        StringBuffer userId, newFileName;
+        newFileName.set(oldFileName);
         if (newFileName.length() < 17)
-            throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid FileId: cannot read FileName.");
-        newFileName.setLength(newFileName.length() - 16); //".wsdfucreate.tmp"
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid FileId: cannot read FileName from %s.", oldFileName);
+        newFileName.setLength(newFileName.length() - 16); //remove ".wsdfucreate.tmp"
```

Comment:
hard wired 16 not very flexible/readable.
Better to define ".wsdfucreate.tmp" as static const and use strlen here.

Reply:
ok